core/src/main/java/org/apache/iceberg/RemoveSnapshots.java

```diff
@@ -310,6 +323,22 @@ private void deleteDataFiles(Set<String> manifestsToScan, Set<String> manifestsT
     return filesToDelete;
   }
 
+  private Snapshot findNthSnapshot(int snapshotIndex) {
```

Comment:
@yathindranath if we switch to relying on snapshot ids, maybe `SnapshotUtil` can help us (with some extensions, if needed).

Reply:
Not sure if I get it correctly. Is the below understanding correct?
1. Find list of Snapshot Ids with `SnapshotUtil.ancestorIds`.
2. Add snapshot Ids from `index = n` to `list.size` to `idsToRemove`.

Reply:
I think `SnapshotUtil` explores parents to build a list of snapshot ids, so that won't solve the issue @xabriel pointed out. Instead, we can use `SnapshotUtil` to determine last N ids and remove all other snapshots. However, it is just an idea.